retrieval/discovery/gce.go

```diff
@@ -177,13 +179,25 @@ func (gd *GCEDiscovery) refresh() (tg *config.TargetGroup, err error) {
 			addr := fmt.Sprintf("%s:%d", priIface.NetworkIP, gd.port)
 			labels[model.AddressLabel] = model.LabelValue(addr)
 
+			// Tags in GCE are mostly used for networking rules (unlike e.g. AWS EC2 tags).
 			if inst.Tags != nil && len(inst.Tags.Items) > 0 {
 				// We surround the separated list with the separator as well. This way regular expressions
 				// in relabeling rules don't have to consider tag positions.
 				tags := gd.tagSeparator + strings.Join(inst.Tags.Items, gd.tagSeparator) + gd.tagSeparator
 				labels[gceLabelTags] = model.LabelValue(tags)
 			}
 
+			// GCE metadata are key-value pairs similar to AWS EC2 tags.
+			if inst.Metadata != nil && len(inst.Metadata.Items) > 0 {
+				for _, i := range inst.Metadata.Items {
+					if i.Value == nil {
```

Comment:
This is just cautiousness. The Metadata Item Value is a *string, so it may be nil. Right now I have no example why it should be, but I don't want to introduce a nil pointer bug.